python/mdvtools/socketio_upload.py

```
@@ -0,0 +1,714 @@
+        # Start background workers
+        self.worker_thread = threading.Thread(target=self._process_queue, daemon=True)
+        self.worker_thread.start()
+        self.cleanup_stop_event = threading.Event()
+        self.cleanup_thread = threading.Thread(target=self._cleanup_abandoned_uploads, daemon=True)
+        self.cleanup_thread.start()
+
+    def _get_upload_dir(self, file_id: str) -> str:
+        """Constructs a secure directory path for storing file uploads."""
+        safe_file_id = secure_filename(file_id)
+        if safe_file_id != file_id:
+             raise ValueError(f"Invalid file_id format: {file_id}")
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)